server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/IdentitySetUpdatePerformer.scala

```diff
@@ -0,0 +1,146 @@
+    def updated: Map[IdentityId, IdentitySetUpdateResponse] =
+      results.flatMap(result => result match {
+        case success: UpdateSuccess => Some((success.id, IdentitySetUpdateResponse.empty))
+        case _ => None
+      }).toMap
+
+    def notUpdated: Map[UnparsedIdentitySetId, SetError] = results.flatMap(result => result match {
+      case failure: UpdateFailure => Some(failure.id, failure.asSetError)
+      case _ => None
+    }).toMap
+  }
+
+  case class UnparsedIdentitySetId(id: Id) {
+    def serialise: String = id.value
+
+    def parse: Either[IllegalArgumentException, IdentityId] = Try(UUID.fromString(id.value))
+      .toEither
+      .left.map({
+      case e: IllegalArgumentException => e
+      case e => new IllegalArgumentException(e)
+    }).map(uuid => IdentityId(uuid))
+  }
+
+  object IdentityUpdateParseException {
+    def from(errors: collection.Seq[(JsPath, collection.Seq[JsonValidationError])]): IdentityUpdateParseException = {
+      val setError: SetError = errors.head match {
+        case (path, Seq()) => SetError.invalidArguments(SetErrorDescription(s"'$path' property in Identity object is not valid"))
+        case (path, Seq(JsonValidationError(Seq("error.path.missing")))) =>
+          SetError.invalidArguments(SetErrorDescription(s"Missing '$path' property in Identity object"))
+        case (path, Seq(JsonValidationError(Seq(message)))) => SetError.invalidArguments(SetErrorDescription(s"'$path' property in Identity object is not valid: $message"))
+        case (path, _) => SetError.invalidArguments(SetErrorDescription(s"Unknown error on property '$path'"))
+      }
+      IdentityUpdateParseException(setError)
+    }
+  }
+
+  case class IdentityUpdateParseException(setError: SetError) extends IllegalArgumentException
+
+  object IdentityUpdate {
+    private val serverSetProperty: Set[String] = Set("id", "mayDelete", "email")
+    private val assignableProperties: Set[String] = Set("name", "replyTo", "bcc", "textSignature", "htmlSignature")
+    private val knownProperties: Set[String] = assignableProperties ++ serverSetProperty
+
+    def validateProperties(jsObject: JsObject): Either[IdentityUpdateParseException, JsObject] =
+      (jsObject.keys.intersect(serverSetProperty), jsObject.keys.diff(knownProperties)) match {
+        case (_, unknownProperties) if unknownProperties.nonEmpty =>
+          Left(IdentityUpdateParseException(SetError.invalidArguments(
+            SetErrorDescription("Some unknown properties were specified"),
+            Some(Properties.toProperties(unknownProperties.toSet)))))
+        case (specifiedServerSetProperties, _) if specifiedServerSetProperties.nonEmpty =>
+          Left(IdentityUpdateParseException(SetError.invalidArguments(
+            SetErrorDescription("Some server-set properties were specified"),
+            Some(Properties.toProperties(specifiedServerSetProperties.toSet)))))
+        case _ => scala.Right(jsObject)
+      }
+  }
+}
```

Comment:
These properties are redundant with what is declared in https://github.com/apache/james-project/pull/752. Please reuse the properties defined there.

Reply:
different properties. 
- create: accept `email` 
- update: not accept `email`

Reply:
Sure, adapt Quan work to you needs but do not duplicate so much code :-P